examples/with-emailverification-with-otp/api-server/server.ts

```diff
@@ -34,69 +35,31 @@ supertokens.init({
-                                    to: user.email,
-                                    subject: "SuperTokens Demo OTP",
-                                    html: getMessageBody(otp, user.email),
-                                });
-                            } catch {
-                                // We ignore this in the example so the example is still usable by getting the otp from the console
-                            }
-                        }
-                    },
-                },
-            },
+        }),
+        MultiFactorAuth.init({
+            firstFactors: ["thirdparty", "emailpassword"], // This is basically disallows using passwordless to sign in
             override: {
-                apis: (oI) => {
-                    return {
-                        ...oI,
-                        verifyEmailPOST: async (input) => {
-                            if (oI.verifyEmailPOST === undefined) {
-                                throw Error("should not come here");
-                            }
-
-                            // retrieve the otp from input
-                            let otp = input.token;
-
-                            // retrieve the token mapped to the otp if it exists
-                            let superTokensToken = otpToTokenMapping.get(otp);
-                            console.log(superTokensToken);
-                            if (superTokensToken !== undefined) {
-                                // if the mapping exists set the token value in the input object to the retrieved token.
-                                input.token = superTokensToken;
-
-                                // remove the otp and token from the mapping
-                                otpToTokenMapping.delete(otp);
-                            } else {
-                                // If the mapping does not exist return an invalid token error
-                                return {
-                                    status: "EMAIL_VERIFICATION_INVALID_TOKEN_ERROR",
-                                };
-                            }
-
-                            let response = await oI.verifyEmailPOST(input);
-                            return response;
-                        },
-                    };
-                },
+                functions: (oI) => ({
+                    ...oI,
+                    getMFARequirementsForAuth: async ({ user }) => (user.loginMethods[0].verified ? [] : ["otp-email"]),
+                }),
             },
         }),
+        AccountLinking.init({
+            shouldDoAutomaticAccountLinking: async (_newAccountInfo, _user, session) =>
+                session
+                    ? {
+                          shouldAutomaticallyLink: true,
+                          shouldRequireVerification: false,
+                      }
+                    : {
+                          shouldAutomaticallyLink: false,
```

Comment:
Changes in the backend SDK:
- Change passwordless create code to not make the session user primary
- And then in consumeCode, we can verify the email address of the session user in case the email of that matches, before attempting to make it a primary user or doing linking (even if we may not be linking the session user to the passwordless one). 

This we can always return true true.